tests/Dfc.CourseDirectory.Testing/TestData/TestData.CreateApprenticeshipUpload.cs

```diff
@@ -0,0 +1,303 @@
+                apprenticeshipWebpage,
+                contactEmail,
+                contactPhone,
+                contactUrl,
+                deliveryMode,
+                venue,
+                yourVenueReference,
+                radius,
+                nationalDelivery,
+                subRegion,
+                errors);
+                _records.Add(record);
+                return this;
+            }
+
+            public ApprenticeshipUploadRowBuilder AddValidRow()
+            {
+                var record = CreateValidRecord("Valid Apprenticeship");
+                _records.Add(record);
+                return this;
+            }
+
+            public ApprenticeshipUploadRowBuilder AddValidRows(int count)
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    AddValidRow();
+                }
+
+                return this;
+            }
+
+            internal IReadOnlyCollection<UpsertApprenticeshipUploadRowsRecord> GetUpsertQueryRows() => _records;
+
+            private UpsertApprenticeshipUploadRowsRecord CreateRecord(
+                Guid apprenticeshipId,
+                int standardCode,
+                int standardVersion,
+                string apprenticeshipInformation,
+                string apprenticeshipWebpage,
+                string contactEmail,
+                string contactPhone,
+                string contactUrl,
+                string deliveryMode,
+                string venue,
+                string yourVenueReference,
+                string radius,
+                string nationalDelivery,
+                string subRegion,
+                IEnumerable<string> errors = null)
+            {
+                var errorsArray = errors?.ToArray() ?? Array.Empty<string>();
+                var isValid = !errorsArray.Any();
+
+                return new UpsertApprenticeshipUploadRowsRecord()
+                {
+                    RowNumber = _records.Count + 2,
+                    IsValid = isValid,
+                    ApprenticeshipId = apprenticeshipId,
+                    StandardCode = standardCode,
+                    StandardVersion = standardVersion,
+                    ApprenticeshipInformation = apprenticeshipInformation,
+                    ApprenticeshipWebpage = apprenticeshipWebpage,
+                    ContactEmail = contactEmail,
+                    ContactPhone = contactPhone,
+                    ContactUrl = contactUrl,
+                    DeliveryMode = deliveryMode,
+                    Venue = venue,
+                    YourVenueReference = yourVenueReference,
+                    Radius = radius,
+                    NationalDelivery = nationalDelivery,
+                    SubRegion = subRegion,
+                    Errors = errors
+                };
+            }
+
+            private UpsertApprenticeshipUploadRowsRecord CreateValidRecord(string apprenticeshipInformation)
```

Comment:
I think this needs to take a standard code + version, rather than apprenticeshipInformation

Reply:
The corresponding method for courses takes a `LearnAimRef` - we have to take that because it needs to exist in the DB to be valid so this method itself can't just invent one. I suspect it'll need to be similar for apprenticeships.